src/org/omegat/core/CoreEvents.java

```diff
@@ -111,9 +111,12 @@ public static void registerEditorEventListener(final IEditorEventListener listen
     public static void unregisterEditorEventListener(final IEditorEventListener listener) {
         EDITOR_EVENT_LISTENERS.remove(listener);
     }
+    
+    public static boolean isInProjectEventLoop = false;
 
     /** Fire event. */
     public static void fireProjectChange(final IProjectEventListener.PROJECT_CHANGE_TYPE eventType) {
+        isInProjectEventLoop = true; // done by the calling thread, not by UI thread (inside invokeLater) !
```

Comment:
The `CoreEvent` class should not have a state in design. The class is entry point and handles it though `IProjectEventListener`. 
Please consider update the interface to support status.